#! /bin/sh /usr/share/dpatch/dpatch-run
# Description: Increase default timeout to ensure we don't hit it
#  In some cases, the 60 second client timeout has been shown to be too
#  low, resulting in ifupdown declaring an interface as 'up' when in fact
#  the dhclient has simply backgrounded to continue trying.  This is
#  addressed in ifupdown by calling dhclient3 with -1 (exit non-zero
#  after the timeout), but we don't ever want to *reach* that timeout:
#  having an interface fail to come up and never be retried is only
#  slightly better than having ifup say the interface is up before it is.
#  Now that ifup's dhclient handling is both reliable and highly
#  parallelized, we can afford to wait longer to be sure we don't hit
#  the timeout when DHCP is really working, so do this by default.
# Author: Steve Langasek <steve.langasek@ubuntu.com>
# Bug-Ubuntu: https://launchpad.net/bugs/838968

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' isc-dhcp~/client/clparse.c isc-dhcp/client/clparse.c
--- isc-dhcp~/client/clparse.c	2011-09-02 14:08:47.220222989 -0700
+++ isc-dhcp/client/clparse.c	2011-09-02 14:13:52.201999111 -0700
@@ -122,7 +122,7 @@
 	memset (&top_level_config, 0, sizeof top_level_config);
 
 	/* Set some defaults... */
-	top_level_config.timeout = 60;
+	top_level_config.timeout = 300;
 	top_level_config.select_interval = 0;
 	top_level_config.reboot_timeout = 10;
 	top_level_config.retry_interval = 300;
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' isc-dhcp~/client/dhclient.conf isc-dhcp/client/dhclient.conf
--- isc-dhcp~/client/dhclient.conf	2011-09-02 14:05:11.000000000 -0700
+++ isc-dhcp/client/dhclient.conf	2011-09-02 14:12:53.751658616 -0700
@@ -6,7 +6,7 @@
 request subnet-mask, broadcast-address, time-offset, routers,
 	domain-name, domain-name-servers, host-name;
 require subnet-mask, domain-name-servers;
-timeout 60;
+timeout 300;
 retry 60;
 reboot 10;
 select-timeout 5;
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' isc-dhcp~/client/dhclient.conf.5 isc-dhcp/client/dhclient.conf.5
--- isc-dhcp~/client/dhclient.conf.5	2011-09-02 14:05:11.000000000 -0700
+++ isc-dhcp/client/dhclient.conf.5	2011-09-02 14:13:27.521856391 -0700
@@ -74,7 +74,7 @@
 statement determines the amount of time that must pass between the
 time that the client begins to try to determine its address and the
 time that it decides that it's not going to be able to contact a
-server.   By default, this timeout is sixty seconds.   After the
+server.   By default, this timeout is 300 seconds.   After the
 timeout has passed, if there are any static leases defined in the
 configuration file, or any leases remaining in the lease database that
 have not yet expired, the client will loop through these leases
@@ -669,7 +669,7 @@
 
 .nf
 
-timeout 60;
+timeout 300;
 retry 60;
 reboot 10;
 select-timeout 5;
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' isc-dhcp~/doc/ja_JP.eucJP/dhclient.conf.5 isc-dhcp/doc/ja_JP.eucJP/dhclient.conf.5
--- isc-dhcp~/doc/ja_JP.eucJP/dhclient.conf.5	2011-09-02 14:05:11.000000000 -0700
+++ isc-dhcp/doc/ja_JP.eucJP/dhclient.conf.5	2011-09-02 14:10:30.840824600 -0700
@@ -77,7 +77,7 @@
 文は、クライアントがアドレスを決める試みを開始してから、
 サーバにアクセスすることが
 できないと判断するまでに経過すべき時間を決めます。
-デフォルトではこのタイムアウト値は 60 秒です。
+デフォルトではこのタイムアウト値は 300 秒です。
 このタイムアウト値が過ぎた後は、
 もし静的なリースが設定ファイルに定義されているか、
 リースデータベースにまだ期限切れになっていないリースが残っていれば、
@@ -583,7 +583,7 @@
 
 .nf
 
-timeout 60;
+timeout 300;
 retry 60;
 reboot 10;
 select-timeout 5;
